### PR TITLE
OSDOCS-6130: Added IMDS support

### PR DESCRIPTION
**Pull request by EricPonvelle**
Version(s):
`enterprise-4.13+`

Issue:
[OSDOCS-6130 ](https://issues.redhat.com/browse/OSDOCS-6130)

Link to docs preview:

* [Cluster Admin](https://60791--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes.html#rosa-imds-machine-pools_rosa-managing-worker-nodes) topic

QE review:
- [ ] QE has approved this change.

Additional information:
Added content about Instance Metadata Service

### CODE REVIEW

File: modules/rosa-imds-machine-pools-ui.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+* You installed and configured the latest AWS (`aws`), ROSA (`rosa`), and OpenShift (`oc`) CLIs on your workstation.
+* You logged in to your Red Hat account by using the `rosa` CLI.
+* You have the permissions to create and manage clusters.
+* You have access to {cluster-manager-url}.
+
+.Procedure
+
+. Log in to the web console.
+. Create a ROSA cluster using your preferences.
+. On the *Default machine pool* page, under the *Instance Metadata Service (IMDS)* section, select whether your machine pools use either both IMDSv1 and IMDSv2 or if you only want to use IMDSv2.
```
Comment (mareklibra) [inline modules/rosa-imds-machine-pools-ui.adoc:22]: This is slightly ambiguous - the `Default machine pool` is part of the previous step, it happens during the step `Create a ROSA cluster...`, not after as one could understand.

File: modules/rosa-imds-machine-pools.adoc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Module included in the following assemblies:
+//
+// * rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes.adoc
+
+:_content-type: CONCEPT
+[id="rosa-imds-machine-pools_{context}"]
+= Instance Metadata Service on machine pools
+
+Instance metadata allows you to configure and manage a running instance. Instance metadate also allows you to access user data that you need to launch your instance.
```
Comment (mareklibra) [inline modules/rosa-imds-machine-pools.adoc:9]: Honestly, I am not sure what practical information a reader can get from these two sentences. I suggest to either skip it or provide a link to AWS documentation (preferred) or rephrase the way that  someone without context knowledge can understand what the metadata or IMD is and what is it useful for.

Comment (mareklibra) [inline modules/rosa-imds-machine-pools.adoc:9]: `metadate` --> `metadata`

File: modules/rosa-imds-machine-pools.adoc [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+IMDSv2 uses session-oriented requests. With session-oriented requests, you create a session token that defines the session duration, which can be a minimum of one second and a maximum of six hours. During the specified duration, you can use the same session token for subsequent requests. After the specified duration expires, you must create a new session token to use for future requests.
+
+There are two types of ways to access instance metadata from a running instance:
```
Comment (mareklibra) [inline modules/rosa-imds-machine-pools.adoc:13]: What about swapping the order of paragraphs here? List the choices first and then provide IMDSv2 description.

File: modules/rosa-imds-machine-pools.adoc [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+* Instance Metadata Service Version 1 (IMDSv1) - a request/response method
+* Instance Metadata Service Version 2 (IMDSv2) - a session-oriented method
+
+When creating your ROSA cluster, you select to use either both IMDSv1 and IMDSv2 or specify that your cluster should only use IMDSv2. The instance metadata service distinguishes between IMDSv1 and IMDSv2 requests based on whether, for any given request, either the PUT or GET headers, which are unique to IMDSv2, are present in that request. If you specify to use IMDSv2 only, IMDSv1 ceases to function for your cluster. All machine pools on your cluster will use whichever IMDS type you select.
```
Comment (mareklibra) [inline modules/rosa-imds-machine-pools.adoc:18]: I would farther clarify what the service is, how to reach it and where the mentioned "requests" and being sent to. It's purely AWS thing.

File: modules/rosa-imds-machine-pools.adoc [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+//
+// * rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes.adoc
+
+:_content-type: CONCEPT
+[id="rosa-imds-machine-pools_{context}"]
+= Instance Metadata Service on machine pools
+
+Instance metadata allows you to configure and manage a running instance. Instance metadate also allows you to access user data that you need to launch your instance.
+
+IMDSv2 uses session-oriented requests. With session-oriented requests, you create a session token that defines the session duration, which can be a minimum of one second and a maximum of six hours. During the specified duration, you can use the same session token for subsequent requests. After the specified duration expires, you must create a new session token to use for future requests.
```
Comment (mareklibra) [inline modules/rosa-imds-machine-pools.adoc:11]: Related to another comment here - some hint about composing "the requests" would be nice.

**Review by mareklibra — CHANGES_REQUESTED** [review]

File: modules/rosa-imds-machine-pools-ui.adoc (hunk 1 above)
Comment (EricPonvelle) [reply modules/rosa-imds-machine-pools-ui.adoc]: @mareklibra The title "Default Machine Pool" is the largest header on this step page. It's how I've seen us refer to this page in [other parts of the documentation](https://docs.openshift.com/rosa/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.html#rosa-sts-creating-cluster-customizations-ocm_rosa-sts-creating-a-cluster-with-customizations) (see step 6). If you'd like, I can call this "On the **Default machine pool** page of the **Cluster settings** step."

image

Comment (EricPonvelle) [reply modules/rosa-imds-machine-pools-ui.adoc]: I updated the text to say:

> . In the **Create a ROSA Cluster** wizard on the **Cluster settings** -> **Machine pool** page, under the *Instance Metadata Service (IMDS)* section, select whether your machine pools use either both IMDSv1 and IMDSv2 or if you only want to use IMDSv2.

File: modules/rosa-imds-machine-pools.adoc (hunk 4 above)
Comment (EricPonvelle) [reply modules/rosa-imds-machine-pools.adoc:16]: Could you explain these pieces?

**Review by EricPonvelle — COMMENTED** [review]
@mareklibra I updated the PR and I had a few items that needed clarification.

File: modules/rosa-imds-machine-pools.adoc (hunk 4 above)
Comment (mareklibra) [reply modules/rosa-imds-machine-pools.adoc:16]: I think the later added link to the AWS documentation is enough. The point was in giving a guiding how to use the metadata and that the service is not part of the cluster.

File: modules/rosa-imds-machine-pools.adoc (hunk 5 above)
Comment (mareklibra) [reply modules/rosa-imds-machine-pools.adoc]: Per other comment, the "hint" is the AWS documentation. Good enough, imho.

**Review by mareklibra — APPROVED** [review]

**Review by skrthomas — COMMENTED** [review]
One tiny nit, otherwise LGTM!

**Pull request merged**